### PR TITLE
media-plugins/gst-plugins-aom: New package

### PR DESCRIPTION
**Pull request by lanodan**
Signed-off-by: Haelwenn (lanodan) Monnier <contact@hacktivis.me>

Set maintainer to gstreamer@ (which I watch on b.g.o btw) but could also take the maintenance of it.

### CODE REVIEW

**Comment by scardracs** [comment]
Is this package required by another package already present in gentoo repo? If not, please mind move it to https://github.com/gentoo/guru

**Comment by lanodan** [comment]
> Is this package required by another package already present in gentoo repo? If not, please mind move it to https://github.com/gentoo/guru

It's not into a `*DEPEND` of one but it's used by gstreamer players and webkit-gtk for playing AV1 videos and it's only a splitted package of gst-plugins-bad that I forgot to include in #19976. Would probably be more annoying to have it in guru when doing version bumps.

File: media-plugins/gst-plugins-aom/gst-plugins-aom-1.18.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+GST_ORG_MODULE=gst-plugins-bad
+
+inherit gstreamer-meson
+
+DESCRIPTION="Alliance for Open Media AV1 plugin for GStreamer"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (scardracs) [inline media-plugins/gst-plugins-aom/gst-plugins-aom-1.18.4.ebuild:11]: Empty IUSE should be dropped

**Review by juippis — COMMENTED** [review]
I'm getting, 
```
 * QA: IUSE=orc is missing while plugin supports it
 * QA: IUSE=introspection is missing while plugin supports it
 * QA: IUSE=orc is missing while plugin supports it
 * QA: IUSE=introspection is missing while plugin supports it
 * QA: depend -> needed comparison:
 *  dev-libs/glib
 * -media-libs/gst-plugins-bad
 * +media-libs/gst-plugins-base
 *  media-libs/gstreamer
 *  media-libs/libaom
```

**Comment by lanodan** [comment]
Fixed for orc and introspection, not sure where the depend/needed comparison one comes from but it seems slightly off to me, it comes from `eclass/gstreamer-meson.eclass` so off-topic for this PR.

**Comment by juippis** [comment]
> not sure where the depend/needed comparison one comes from but it seems slightly off to me, it comes from `eclass/gstreamer-meson.eclass` anyway so off-topic for this PR.

It's a QA check script by @ionenwks and it basically checks your packages vdb's NEEDED and compares it to other packages vdb's PROVIDED. See:
```
# cat /var/db/pkg/media-plugins/gst-plugins-aom-1.18.4/NEEDED
/usr/lib/gstreamer-1.0/libgstaom.so libgstvideo-1.0.so.0,libgstreamer-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libaom.so.3,libc.so.6
/usr/lib64/gstreamer-1.0/libgstaom.so libgstvideo-1.0.so.0,libgstreamer-1.0.so.0,libgobject-2.0.so.0,libglib-2.0.so.0,libaom.so.3,libc.so.6
```
```
# grep libgstvideo /var/db/pkg/*/*/PROVIDES
/var/db/pkg/media-libs/gst-plugins-bad-1.18.4/PROVIDES:x86_32: libgstaccurip.so libgstadaptivedemux-1.0.so.0 libgstadpcmdec.so libgstadpcmenc.so libgstaiff.so libgstasfmux.so libgstaudiobuffersplit.so libgstaudiofxbad.so libgstaudiolatency.so libgstaudiomixmatrix.so libgstaudiovisualizers.so libgstautoconvert.so libgstbadaudio-1.0.so.0 libgstbasecamerabinsrc-1.0.so.0 libgstbayer.so libgstcamerabin.so libgstcodecparsers-1.0.so.0 libgstcodecs-1.0.so.0 libgstcoloreffects.so libgstdebugutilsbad.so libgstdvbsubenc.so libgstdvbsuboverlay.so libgstdvdspu.so libgstfaceoverlay.so libgstfestival.so libgstfieldanalysis.so libgstfreeverb.so libgstfrei0r.so libgstgaudieffects.so libgstgdp.so libgstgeometrictransform.so libgstid3tag.so libgstinsertbin-1.0.so.0 libgstinter.so libgstinterlace.so libgstipcpipeline.so libgstisoff-1.0.so.0 libgstivfparse.so libgstivtc.so libgstjp2kdecimator.so libgstjpegformat.so libgstlegacyrawparse.so libgstmidi.so libgstmpegpsdemux.so libgstmpegpsmux.so libgstmpegts-1.0.so.0 libgstmpegtsdemux.so libgstmpegtsmux.so libgstmxf.so libgstnetsim.so libgstpcapparse.so libgstphotography-1.0.so.0 libgstplayer-1.0.so.0 libgstpnm.so libgstproxy.so libgstremovesilence.so libgstrist.so libgstrtmp2.so libgstrtpmanagerbad.so libgstrtponvif.so libgstsctp-1.0.so.0 libgstsdpelem.so libgstsegmentclip.so libgstshm.so libgstsiren.so libgstsmooth.so libgstspeed.so libgstsubenc.so libgstswitchbin.so libgsttimecode.so libgsttranscoder-1.0.so.0 libgsturidownloader-1.0.so.0 libgstvideofiltersbad.so libgstvideoframe_audiolevel.so libgstvideoparsersbad.so libgstvideosignal.so libgstvmnc.so libgstwebrtc-1.0.so.0 libgsty4mdec.so x86_64: libgstaccurip.so libgstadaptivedemux-1.0.so.0 libgstadpcmdec.so libgstadpcmenc.so libgstaiff.so libgstasfmux.so libgstaudiobuffersplit.so libgstaudiofxbad.so libgstaudiolatency.so libgstaudiomixmatrix.so libgstaudiovisualizers.so libgstautoconvert.so libgstbadaudio-1.0.so.0 libgstbasecamerabinsrc-1.0.so.0 libgstbayer.so libgstcamerabin.so libgstcodecparsers-1.0.so.0 libgstcodecs-1.0.so.0 libgstcoloreffects.so libgstdebugutilsbad.so libgstdvbsubenc.so libgstdvbsuboverlay.so libgstdvdspu.so libgstfaceoverlay.so libgstfestival.so libgstfieldanalysis.so libgstfreeverb.so libgstfrei0r.so libgstgaudieffects.so libgstgdp.so libgstgeometrictransform.so libgstid3tag.so libgstinsertbin-1.0.so.0 libgstinter.so libgstinterlace.so libgstipcpipeline.so libgstisoff-1.0.so.0 libgstivfparse.so libgstivtc.so libgstjp2kdecimator.so libgstjpegformat.so libgstlegacyrawparse.so libgstmidi.so libgstmpegpsdemux.so libgstmpegpsmux.so libgstmpegts-1.0.so.0 libgstmpegtsdemux.so libgstmpegtsmux.so libgstmxf.so libgstnetsim.so libgstpcapparse.so libgstphotography-1.0.so.0 libgstplayer-1.0.so.0 libgstpnm.so libgstproxy.so libgstremovesilence.so libgstrist.so libgstrtmp2.so libgstrtpmanagerbad.so libgstrtponvif.so libgstsctp-1.0.so.0 libgstsdpelem.so libgstsegmentclip.so libgstshm.so libgstsiren.so libgstsmooth.so libgstspeed.so libgstsubenc.so libgstswitchbin.so libgsttimecode.so libgsttranscoder-1.0.so.0 libgsturidownloader-1.0.so.0 libgstvideofiltersbad.so libgstvideoframe_audiolevel.so libgstvideoparsersbad.so libgstvideosignal.so libgstvmnc.so libgstwebrtc-1.0.so.0 libgsty4mdec.so
/var/db/pkg/media-libs/gst-plugins-base-1.18.4/PROVIDES:x86_32: libgstadder.so libgstallocators-1.0.so.0 libgstapp-1.0.so.0 libgstapp.so libgstaudio-1.0.so.0 libgstaudioconvert.so libgstaudiomixer.so libgstaudiorate.so libgstaudioresample.so libgstaudiotestsrc.so libgstcompositor.so libgstencoding.so libgstfft-1.0.so.0 libgstgio.so libgstgl-1.0.so.0 libgstogg.so libgstopengl.so libgstoverlaycomposition.so libgstpango.so libgstpbtypes.so libgstpbutils-1.0.so.0 libgstplayback.so libgstrawparse.so libgstriff-1.0.so.0 libgstrtp-1.0.so.0 libgstrtsp-1.0.so.0 libgstsdp-1.0.so.0 libgstsubparse.so libgsttag-1.0.so.0 libgsttcp.so libgsttypefindfunctions.so libgstvideo-1.0.so.0 libgstvideoconvert.so libgstvideorate.so libgstvideoscale.so libgstvideotestsrc.so libgstvolume.so libgstvorbis.so libgstximagesink.so libgstxvimagesink.so x86_64: libgstadder.so libgstallocators-1.0.so.0 libgstapp-1.0.so.0 libgstapp.so libgstaudio-1.0.so.0 libgstaudioconvert.so libgstaudiomixer.so libgstaudiorate.so libgstaudioresample.so libgstaudiotestsrc.so libgstcompositor.so libgstencoding.so libgstfft-1.0.so.0 libgstgio.so libgstgl-1.0.so.0 libgstogg.so libgstopengl.so libgstoverlaycomposition.so libgstpango.so libgstpbtypes.so libgstpbutils-1.0.so.0 libgstplayback.so libgstrawparse.so libgstriff-1.0.so.0 libgstrtp-1.0.so.0 libgstrtsp-1.0.so.0 libgstsdp-1.0.so.0 libgstsubparse.so libgsttag-1.0.so.0 libgsttcp.so libgsttypefindfunctions.so libgstvideo-1.0.so.0 libgstvideoconvert.so libgstvideorate.so libgstvideoscale.so libgstvideotestsrc.so libgstvolume.so libgstvorbis.so libgstximagesink.so libgstxvimagesink.so
```
and you'll see that for example *libgstvideo-1.0.so.0*, which is NEEDED by gst-plugins-aom, comes from **gst-plugins-base** instead of **gst-plugins-bad**.

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**